presto-tpch/src/main/java/com/facebook/presto/tpch/TpchRecordSet.java

```diff
@@ -200,32 +208,35 @@ public void close()
 
         private boolean rowMatchesPredicate()
         {
-            if (!predicate.isPresent()) {
+            if (!predicate.isPresent() || predicate.get().isAll()) {
                 return true;
             }
-            return predicate.get().test(rowMap());
-        }
+            Set<ColumnHandle> predicatePushdownColumns = predicate.get().getDomains().get().keySet();
+            Map<ColumnHandle, NullableValue> rowMap = predicatePushdownColumns.stream()
+                    .collect(toImmutableMap(
+                            column -> column,
+                            column -> {
+                                TpchColumnHandle tpchColumnHandle = (TpchColumnHandle) column;
+                                Type type = tpchColumnHandle.getType();
+                                TpchColumn tpchColumn = table.getColumn(tpchColumnHandle.getColumnName());
+                                return NullableValue.of(type, getPrestoObject(tpchColumn, type));
+                            }));
 
-        private Map<ColumnHandle, NullableValue> rowMap()
-        {
-            ImmutableMap.Builder<ColumnHandle, NullableValue> builder = ImmutableMap.builder();
-            for (int field = 0; field < columnHandles.size(); ++field) {
-                Type type = columnTypes.get(field);
-                builder.put(columnHandles.get(field), NullableValue.of(type, getPrestoObject(field, type)));
-            }
-            return builder.build();
+            TupleDomain<ColumnHandle> rowTupleDomain = TupleDomain.fromFixedValues(rowMap);
+
+            return predicate.get().contains(rowTupleDomain);
```

Comment:
Previously the `.test()` method was used. Are you sure `.contains()` is correct? `TupleDomain::contains` caused the problem in #9839, see https://github.com/prestodb/presto/pull/9839/commits/d4a0f2d9f03f1024449a5319a87edef549286616#diff-ccd5d11614e67178cf1dfe85be861160L411

Reply:
Yes. See the response for https://github.com/prestodb/presto/pull/9823/commits/bda705b783589b31fd86c20283c0782ade6f7164#r165374015